components/TextEditor/TextEditorContainer.tsx

```diff
@@ -24,29 +22,41 @@ const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({heigh
     const [error, setError] = useState("");
     const getEditorContentIfMountedRef: MutableRefObject<EditorContentGetter> = useRef(() => "");
     const router=useRouter();
-    
+
     function handleIsEditorMounted(_getEditorContents: EditorContentGetter): void {
         setIsEditorReady(true);
+        if(isEditorReady) {
         getEditorContentIfMountedRef.current = _getEditorContents;
-     }
+     }}
 
-   
-    function submitForm() {
-        const testAnswer=getEditorContentIfMountedRef.current();
-        addTestSubmisson(token,{testId,testAnswer})
-            .then((response)=>{  
+
+    function submitForm(): void {
+        const testAnswer = getEditorContentIfMountedRef.current();
+        addTestSubmission(token,{testId,testAnswer})
+            .then((response)=>{
                 if (response.ok) {
-                    router.push(`/submitted?token=${token}`);
+                    router.push('/testlibrary').then(r => r.valueOf());
```

Comment:
this will break because it doesn't have the token.
Also, whats `then(r => r.valueOf()` there for?